commons-core/src/main/scala/com/avsystem/commons/serialization/cbor/RawCbor.scala

```diff
@@ -1,17 +1,42 @@
 package com.avsystem.commons
 package serialization.cbor
 
-import com.avsystem.commons.serialization.{GenCodec, TypeMarker}
+import com.avsystem.commons.serialization.{GenCodec, SizePolicy, TypeMarker}
+
+import scala.annotation.tailrec
 
 final case class RawCbor(bytes: Array[Byte], offset: Int, length: Int) {
   require(offset >= 0 && length >= 0 && offset + length <= bytes.length)
 
+  override def equals(other: Any): Boolean = other match {
+    case RawCbor(otherBytes, otherOffset, otherLength) =>
+      @tailrec def loop(i: Int): Boolean =
+        i >= length || bytes(offset + i) == otherBytes(otherOffset + i) && loop(i + 1)
+      length == otherLength && loop(0)
+    case _ => false
+  }
+
+  override lazy val hashCode: Int = {
+    @tailrec def loop(i: Int, acc: Int): Int =
+      if (i >= offset + length) acc
+      else loop(i + 1, acc * 31 + bytes(i))
```

Comment:
how will it behave on integer overflow? It is not specified by JVM IIRC

Reply:
It will overflow. So what? This is literally the same as `java.util.Arrays#hashCode(byte[])` but I can't use it because it doesn't accept offsets.